test/location.test.js

```diff
@@ -104,6 +145,28 @@ define(function(require) {
 
           location.on('statechange.test-no-dispatch', null)
         })
+
+        it('should preserve query parameters', function() {
+          var loc = window.location
+          var originalQuery = new URLSearchParams(loc.search.substring(1))
+
+          originalQuery.set('foo', 'bar')
+          originalQuery.set('fizz', 'buzz')
+
+          window.history.replaceState(
+            null, null,
+            loc.origin + loc.pathname + '?' + originalQuery.toString()
```

Comment:
Instead of all this, can we just have `/?foo=bar&fizz=buzz' or something? It's a bit hard to follow this test.